app/Authentication/src/Http/Controllers/AdminAuthController.php

```diff
@@ -27,13 +30,28 @@ public function index(): AuthResource
     #[Group('Authentication')]
     #[BodyParam('email', 'password', required: true)]
 
-    public function login(AdminLoginRequest $request): AuthResource
+    public function login(AdminLoginRequest $request)
     {
         $user = User::where('email', $request->email)->first();
 
-        $request->authenticate($user);
+        $credentials = request(['email', 'password']);
 
-        return new AuthResource($user);
+        if (!$user || !$user->hasRole(RoleName::ADMIN) || ! $token = auth()->attempt($credentials)) {
+            throw ValidationException::withMessages([
+                'email' => __('auth.failed'),
+            ]);
+        }
+
+
+        return  response()->json([
```

Comment:
You can still use a AuthResource which will be only used on login / refresh token, but you could also set the data from the outside and combine it inside by using the "with" or "additional" methods on the resource class.

More info here: https://laravel.com/docs/11.x/eloquent-resources